compiler/lcalc/compile_with_exceptions.ml

```diff
@@ -64,8 +64,10 @@ and translate_expr (ctx : 'm ctx) (e : 'm D.expr) : 'm A.expr boxed =
     Expr.ecatch (translate_expr ctx arg) EmptyError
       (Expr.eraise NoValueProvided m)
       m
-  | EDefault { excepts = [exn]; just; cons } when !Cli.optimize_flag ->
-    (* FIXME: bad place to rely on a global flag *)
+  | EDefault { excepts = [exn]; just; cons }
+  (* Specific optimisation for this case *) ->
+    (* FIXME: this case used to be disabled when optimisations are disabled, but
```

Comment:
Like this FIXE here

Reply:
Ah right here I am not sure what is best: I'd like to keep the case unconditionally (without checking for the optimisations flag) but it's also possible to pass the flag along if it makes sense.

Subsidiary question: if we enable optimisations in `lcalc`, doesn't this term get rewritten in the same way by the following optimisation pass ? In which case just removing this case would yield the same results as before (both with and without `-O`)

Reply:
> In which case just removing this case would yield the same results as before (both with and without -O)

Yeah you're right, let's just delete this code!